homeassistant/components/mqtt/device_tracker.py

```diff
@@ -127,6 +132,8 @@ def message_received(msg: ReceiveMessage) -> None:
                 self._location_name = STATE_HOME
             elif payload == self._config[CONF_PAYLOAD_NOT_HOME]:
                 self._location_name = STATE_NOT_HOME
+            elif payload == self._config[CONF_PAYLOAD_AUTO_ZONE]:
+                self._location_name = None
```

Comment:
As you can see the added code is not covered by tests. It would be nice if we could assert on the automatic assigned zone when we supply `lat` and `lon` coordinates.

Existing tests can be found here:

https://github.com/home-assistant/core/blob/9aaeefeb8e76d9c56b7bd32c300c3149b3ba0c22/tests/components/mqtt/test_device_tracker.py#L351-L442